packages/rich-text/src/component/index.native.js

```diff
@@ -631,29 +630,6 @@ export class RichText extends Component {
 		}
 	}
 
-	willTrimSpaces( html ) {
-		const { tagName } = this.props;
-
-		// aztec won't trim spaces in a case of <pre> block, so we are excluding it
-		if ( tagName === 'pre' ) {
-			return false;
-		}
-
-		// regex for detecting spaces around block element html tags
-		const blockHtmlElements =
-			'(div|br|blockquote|ul|ol|li|p|pre|h1|h2|h3|h4|h5|h6|iframe|hr)';
-		const leadingOrTrailingSpaces = new RegExp(
-			`(\\s+)<\/?${ blockHtmlElements }>|<\/?${ blockHtmlElements }>(\\s+)`,
```

Comment:
👋 @mchowning, looks like [the whitespace preserving code on the Aztec side](https://github.com/wordpress-mobile/AztecEditor-Android/blob/1e09b7daa88498c2c8d62b09cee0c8af8bb86616/aztec/src/main/java/org/wordpress/aztec/Html.java#L732) is preserving simple spaces only, while the regex here is checking for all whitespace chars. I'm under the impression that the Aztec side behavior won't actually preserve all whitespace chars, so, can you check as well?

I tried having a tab char (ASCII 0x09) leading and trialing the test but couldn't break the editor (the whitespace was preserved) although I'm not sure I did actually manage to insert the TAB, it might actually had become a regular space char.

Reply:
> looks like the whitespace preserving code on the Aztec side is preserving simple spaces only, while the regex here is checking for all whitespace chars.

Are you aware of any specific reason the regex is checking for all whitespace instead of just spaces. I wasn't able to find a reason that was needed (both [the comment](https://github.com/WordPress/gutenberg/pull/22006/files#diff-dda065334d6da1eb4149443ac3439f7eL642) on the regex and [the original change with adding this regex](https://github.com/WordPress/gutenberg/pull/15021) seemed to only involve issues with space characters specifically.

> I'm under the impression that the Aztec side behavior won't actually preserve all whitespace chars, so, can you check as well?
>
> I tried having a tab char (ASCII 0x09) leading and trialing the test but couldn't break the editor (the whitespace was preserved) although I'm not sure I did actually manage to insert the TAB, it might actually had become a regular space char.

I noticed the same thing, a lot of non-standard characters I tried (tab, line feed, carriage return, non-breaking space) were converted into spaces (well, eventually: line feed would briefly be a newline and non-breaking space would briefly be `&nbsp;`, but those would then get turned into spaces). In everything I tried, I couldn't find anything that broke on Gutenberg.

Reply:
> Are you aware of any specific reason the regex is checking for all whitespace instead of just spaces.

If I recall correctly, the whitespace in general gets removed by Aztec because HTML itself is ignoring them, and Aztec tries to follow that logic. In this case, I think I tried to generalize and having in mind that the [AztecParser.tidy() method](https://github.com/wordpress-mobile/AztecEditor-Android/blob/1e09b7daa88498c2c8d62b09cee0c8af8bb86616/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L656-L664) and perhaps other components too will try to remove some whitespace.